server/server.go

```diff
@@ -105,6 +108,28 @@ func handler(s ssh.Session) {
 	verbose("handler exits")
 }
 
+func Advertise(hostname, port, txt string) (*mdns.Server, error) {
+	v("starting mdns server")
+	// Setup our service export
+	host, _ := os.Hostname()
+	p, err := strconv.ParseUint(port, 0, 0)
+	if err != nil {
+		return nil, err
+	}
+	service, err := mdns.NewMDNSService(host, "_ncpu._tcp", "", "", int(p), nil, strings.Split(txt, ","))
+	if err == nil {
+		// Create the mDNS server, defer shutdown
+		// we have to lookup the right interface, this will have to be passed in from command line
+		server, err := mdns.NewServer(&mdns.Config{Zone: service})
+		return server, err
+	} else {
+		log.Printf(`mdns new service failure: %v`, err)
```

Comment:
Do you want to return an error here instead of logging and returning err = nil?

I'm trying to understand how the code is structured - is the goal here to still return a nil error if mdns.NewMDNSService() fails?    If not then maybe you can simplify and restructure the code as:
```
service, err := ....
if err != nil {
  // Handle the error
}
return mdns.NewServer(&mdns.Config{Zone: service})
```
?

Reply:
the original goal was to try and keep mdns as optional as possible so if mdns resolution fails, just fall through and hope the user specified the hostname.  I figured the end result would be a host not found if otherwise -- but I'm happy to take action to make this more explicit or specific (no available resource found versus host not found)

Reply:
Thanks for letting me know the goal, that is helpful.   In this case I would think that it's still better to explicitly return the error here from `Advertise` in server.go (because we are failing to advertise...), but ignore the error in the *caller* of the function, which appears to be cpud/serve.go in the `if *mdnsEnabled{` section of code - and it appears you're already doing that.

So properly returning the errors here won't change the existing optionality that is already accomplished by the caller in cpud/serve.go ignore the returned error.

Please let me know if I've misunderstood? Thanks!

Reply:
I've added back in error handling and made failure less silent if you selected the mdns advertisement option.